front-end-tools/src/components/InitComponent.tsx

```diff
@@ -142,29 +144,25 @@ export default function InitComponent(props: ConnectionProps) {
     }, [connection, client, txHash]);
 
     useEffect(() => {
+        setIsModuleReferenceAlreadyDeployedError(undefined);
         if (connection && client && moduleReference) {
-            if (REG_MODULE_REF.test(moduleReference)) {
-                client
-                    .getModuleSource(ModuleReference.fromHexString(moduleReference))
-                    .then((value) => {
-                        if (value === undefined) {
-                            setIsModuleReferenceAlreadyDeployedStep2(false);
-                        } else {
-                            setIsModuleReferenceAlreadyDeployedStep2(true);
-                        }
-                    })
-                    .catch(() => {
-                        setIsModuleReferenceAlreadyDeployedStep2(false);
-                    });
-            }
+            client
+                .getModuleSource(moduleReference)
+                .then((value) => {
+                    setIsModuleReferenceAlreadyDeployed(value !== undefined);
+                })
+                .catch((e) => {
+                    setIsModuleReferenceAlreadyDeployedError((e as Error).message.replaceAll('%20', ' '));
+                    setIsModuleReferenceAlreadyDeployed(false);
+                });
         }
     }, [connection, client, moduleReference]);
 
     const shouldWarnDifferenceModuleReferences = useMemo(() => {
         if (
             moduleReference !== undefined &&
             moduleReferenceCalculated !== undefined &&
-            moduleReferenceCalculated !== moduleReference
+            moduleReferenceCalculated.moduleRef !== moduleReference.moduleRef
```

Comment:
Alternatively, consider using the following (which is generally available for all our wrapper types):
```suggestion
            ModuleReference.equals(moduleReferenceCalculated, moduleReference)
```